tests/unit/dataactcore/utils/test_tracing.py

```diff
@@ -138,8 +138,8 @@ def test_logging_trace_spans(caplog, capsys):
 
     captured = capsys.readouterr()
     assert test_msg in caplog.text, "caplog.text did not seem to capture logging output during test"
-    assert hex(trace_id) in captured.out, "trace_id not found in logging output"
-    assert hex(span_id) in captured.out, "span_id not found in logging output"
+    assert f'{trace_id:x}' in captured.out, "trace_id not found in logging output"
+    assert f'{span_id:x}' in captured.out, "span_id not found in logging output"
```

Comment:
The `hex` function was not including leading zeros after the `0x` in the hex string.  This was causing this test to randomly fail.  I used python's built in formatter to format this as a string without the leading `0x` in order to address this issue.  See [docs](https://docs.python.org/3/library/functions.html#hex).

Reply:
Great catch! I was so curious about the root cause of that test randomly failing. Thank you.